src/main/java/net/zhuoweizhang/raspberryjuice/RemoteSession.java

```diff
@@ -222,43 +283,123 @@ protected void handleCommand(String c, String[] args) {
 				
 			// events.clear
 			} else if (c.equals("events.clear")) {
-				interactEventQueue.clear();
-				chatPostedQueue.clear();
+				int entityId = -1;
+				if (args[0].length() > 0)
+					entityId = Integer.parseInt(args[0]);
+				if (entityId == -1) {
+					interactEventQueue.clear();
+					chatPostedQueue.clear();
+					projectileHitQueue.clear();
+				}
+				else {
+					for (Iterator<PlayerInteractEvent> iter = interactEventQueue.iterator(); iter.hasNext(); ) {
+						PlayerInteractEvent event = iter.next();
+						if (event.getPlayer().getEntityId() == entityId)
+							iter.remove();
+					}
+					for (Iterator<AsyncPlayerChatEvent> iter = chatPostedQueue.iterator(); iter.hasNext(); ) {
+						AsyncPlayerChatEvent event = iter.next();
+						if (event.getPlayer().getEntityId() == entityId)
+							iter.remove();
+					}
+					for (Iterator<ProjectileHitEvent> iter = projectileHitQueue.iterator(); iter.hasNext(); ) {
+						ProjectileHitEvent event = iter.next();
+						Arrow arrow = (Arrow) event.getEntity();
+						LivingEntity shooter = (LivingEntity)arrow.getShooter();
+						if (shooter.getEntityId() == entityId)
+							iter.remove();
+					}
+				}
 				
 			// events.block.hits
 			} else if (c.equals("events.block.hits")) {
+				int entityId = -1;
+				if (args[0].length() > 0)
+					entityId = Integer.parseInt(args[0]);
 				StringBuilder b = new StringBuilder();
-		 		PlayerInteractEvent event;
-				while ((event = interactEventQueue.poll()) != null) {
-					Block block = event.getClickedBlock();
-					Location loc = block.getLocation();
-					b.append(blockLocationToRelative(loc));
-					b.append(",");
-					b.append(blockFaceToNotch(event.getBlockFace()));
-					b.append(",");
-					b.append(event.getPlayer().getEntityId());
-					if (interactEventQueue.size() > 0) {
+				for (Iterator<PlayerInteractEvent> iter = interactEventQueue.iterator(); iter.hasNext(); ) {
```

Comment:
Again I have a similar opinion. I think if this is going to be entity specific it would be better as a new command placed against the entity rather than changing an existing one.

Reply:
Yes, but this way I made two possibilities within one method. You can pass optional player's entityId to retrieve events of specified player only or you can pass nothing to have old functionality affecting all players. Sample:

import mcpi.minecraft as minecraft
import time
import sys
mc = minecraft.Minecraft.create("127.0.0.1")
player = "MST"
playerEntityId = mc.getPlayerEntityId(player)
while True:
    hits = mc.events.pollBlockHits(playerEntityId)
    for hit in hits:
        print(hit.pos.x, hit.pos.y, hit.pos.z, hit.face, hit.entityId)
        sys.stdout.flush()
    time.sleep(0.1)

Reply:
I understand that it can now be used for 2 purposes, but it changes an existing API and a lower impacting change would be to create a new method.

My preference would be to implement a method against the entity / player so you could do something like.

```python
entity_block_hits = mc.player.pollBlockHits()
entity_block_hits = mc.entity.pollBlockHits(entityId)
```

Leaving the `mc.events` functions as is.